packages/tools/devtools/devtools-core/src/DecomposedContainer.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TypedEventEmitter } from "@fluid-internal/client-utils";
+import type { IAudience } from "@fluidframework/container-definitions";
+import type {
+	AttachState,
+	IContainerEvents,
+	ICriticalContainerError,
+} from "@fluidframework/container-definitions/internal";
+import { ConnectionState } from "@fluidframework/container-loader";
+import type { IContainerRuntime } from "@fluidframework/container-runtime-definitions/internal";
+import type { IEventProvider } from "@fluidframework/core-interfaces";
+
+/**
+ * A lightweight abstraction of a container that provides
+ * only the essential properties and methods needed for Fluid DevTools functionality.
+ */
+export interface DecomposedContainer extends IEventProvider<IContainerEvents> {
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.audience}
+	 */
+	readonly audience: IAudience;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.clientId}
+	 */
+	readonly clientId?: string | undefined;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.attachState}
+	 */
+	readonly attachState: AttachState;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.connectionState}
+	 */
+	readonly connectionState: ConnectionState;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.closed}
+	 */
+	readonly closed: boolean;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.connect}
+	 */
+	connect?(): void;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.disconnect}
+	 */
+	disconnect?(): void;
+
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.close}
+	 */
+	close?(error?: ICriticalContainerError): void;
+}
+
+/**
+ * Implementation of {@link DecomposedContainer} that wraps an {@link @fluidframework/container-runtime-definitions/internal#IContainerRuntime}.
+ * This class provides a bridge between {@link @fluidframework/container-runtime-definitions/internal#IContainerRuntime} and the devtools system by exposing runtime properties and events.
+ */
+export class DecomposedContainerForContainerRuntime
```

Comment:
Nit: since this is specifically used by `ContainerRuntimeDevtools`, it might make more sense for this class to live there.